src/UrqlClient.re

```diff
@@ -9,11 +9,73 @@ let unwrapFetchOptions = fetchOptions =>
   | FetchFn(fn) => fn()
   };
 
+module UrqlExchanges = {
+  [@bs.deriving jsConverter]
+  type operationType = [
+    | [@bs.as "subscription"] `Subscription
+    | [@bs.as "query"] `Query
+    | [@bs.as "mutation"] `Mutation
+    | [@bs.as "teardown"] `Teardown
+  ];
+
+  [@bs.deriving abstract]
+  type operationContext = {
+    [@bs.optional]
+    fetchOptions: Fetch.requestInit,
+    requestPolicy: UrqlTypes.requestPolicy,
+    url: string,
+  };
+
+  [@bs.deriving abstract]
+  type operation = {
+    [@bs.optional]
+    variables: Js.Json.t,
+    key: int,
+    query: string,
+    operationName: operationType,
+    context: operationContext,
+  };
+
+  type operationData;
+
+  [@bs.deriving abstract]
+  type operationResult = {
+    operation,
+    [@bs.optional]
+    data: operationData,
+    [@bs.optional]
+    error: UrqlCombinedError.t,
+  };
+
+  type exchangeIO =
+    Wonka.Types.sourceT(operation) => Wonka.Types.sourceT(operationResult);
+
+  [@bs.deriving abstract]
+  type exchangeInput = {
+    forward: exchangeIO,
+    client: t,
+  };
+
+  type exchange = exchangeInput => exchangeIO;
+
+  [@bs.module "urql"] external cacheExchange: exchange = "";
+  [@bs.module "urql"] external debugExchange: exchange = "";
+  [@bs.module "urql"] external dedupExchange: exchange = "";
+  [@bs.module "urql"] external fallbackExchangeIO: exchangeIO = "";
+  [@bs.module "urql"] external fetchExchange: exchange = "";
+  [@bs.module "urql"] external subscriptionExchange: exchange = "";
```

Comment:
Looking more closely, `subscriptionExchange` may differ slightly in API from the other exchanges. I'll circle back to these bindings in the next PR, which'll add tests!

Reply:
Yea that is a factory that takes some extra options